streams/src/test/java/org/apache/kafka/streams/integration/utils/IntegrationTestUtils.java

```diff
@@ -104,6 +107,40 @@
     public static final long DEFAULT_TIMEOUT = 60 * 1000L;
     private static final Logger LOG = LoggerFactory.getLogger(IntegrationTestUtils.class);
 
+    /**
+     * Repeatedly runs the query until the response is valid and then return the response.
+     * <p>
+     * Validity in this case means that the response contains all the desired partitions or that
+     * it's a global response.
+     * <p>
+     * Once position bounding is generally supported, we should migrate tests to wait on the
+     * expected response position.
+     */
+    public static <R> StateQueryResult<R> iqv2WaitForPartitionsOrGlobal(
```

Comment:
The javadoc pretty much sums this up. It's a harbinger of good things to come, since as you can see, we'll no longer be vulnerable to race conditions like rebalances sneaking in during the test. We can just wait for a valid result and then verify it.